### ISSUE TITLE
Installing under Ubuntu 22.04

### ISSUE THREAD

**Issue by andimik** [deeptho/neumodvb#26]
Ubuntu 22.04 does not have `python-wxgtk3.0`

```
$ LC_ALL=C sudo  apt install -y  libboost-all-dev libgtk-3-0 libgtk-3-dev curl libcurl4-openssl-dev  libwxgtk-media3.0-gtk3-dev gettext libexif-dev libavcodec-dev libavformat-dev libavutil-dev libswscale-dev   python3-configobj python3-cachetools python3-jinja2 python3-pip clang-format python3-sip-dev libconfig-dev libconfig++-dev libdvbcsa-dev  libmpv-dev freeglut3-dev python-wxgtk3.0 python3-wxgtk-media4.0 python3-wxgtk-webview4.0 python3-wxgtk4.0 python3-scipy clang lsb-core lsb-release python3-regex
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package python-wxgtk3.0
E: Couldn't find any package by glob 'python-wxgtk3.0'
```

Therefore, I cannot install `neumodvb`:

```
$ LC_ALL=C cmake ..
-- xxx=/jammy/
-- Modern OS detected - using clang 20
-- Found Python: /usr/bin/python3.10 (found version "3.10.6") found components: Interpreter Development Development.Module Development.Embed 
-- xxx=/jammy/
-- pybind11 v2.10.0 dev1
-- Found Python: /usr/bin/python3.10 (found suitable version "3.10.6", minimum required is "3.7") found components: Interpreter 
CMake Error at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find wxWidgets (missing: wxWidgets_LIBRARIES
  wxWidgets_INCLUDE_DIRS)
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.22/Modules/FindwxWidgets.cmake:1025 (find_package_handle_standard_args)
  src/receiver/CMakeLists.txt:1 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/andreas/apps/neumodvb/build/CMakeFiles/CMakeOutput.log".
```

Any suggestions?

**Comment by andimik** [comment]
Maybe this solves it, but I am not a programmer:

```
$ LC_ALL=C sudo apt-get install -y libwxgtk3.0-gtk3-dev
```

```
$ LC_ALL=C cmake ..
-- xxx=/jammy/
-- Modern OS detected - using clang 20
-- Found Python: /usr/bin/python3.10 (found version "3.10.6") found components: Interpreter Development Development.Module Development.Embed 
-- xxx=/jammy/
-- pybind11 v2.10.0 dev1
-- Found Python: /usr/bin/python3.10 (found suitable version "3.10.6", minimum required is "3.7") found components: Interpreter 
-- Found wxWidgets: -L/usr/lib/x86_64-linux-gnu;-pthread;;;-lwx_baseu_net-3.0;-lwx_gtk3u_gl-3.0;-lwx_gtk3u_core-3.0;-lwx_baseu-3.0 (found version "3.0.5") 
-- Searching for log4cxx/logger.h
-- Searching for liblog4cxx
-- Log4cxx_INCLUDE_DIR (missing: Log4cxx_LIBRARY) 
-- The host system is: Linux-5.15.0-50-generic.
-- Project base dir is: /home/andreas/apps/neumodvb
-- Configuring done
-- Generating done
-- Build files have been written to: /home/andreas/apps/neumodvb/build
```

**Comment by andimik** [comment]
Then, 

```
/home/andreas/apps/neumodvb/src/stackstring/../util/logger.h:24:10: fatal error: 'log4cxx/logger.h' file not found
#include <log4cxx/logger.h>
```
seems to be fixed by

```
sudo apt install liblog4cxx12 liblog4cxx-dev
```
which is not mentioned in the readme.

**Comment by andimik** [comment]
and 

```
$ sudo apt-get install freeglut3-dev freeglut3
```
solves the following compiling error:

```
/home/andreas/apps/neumodvb/src/viewer/neumompv_pybind.cc:26:10: fatal error: 'GL/glut.h' file not found
#include <GL/glut.h>
         ^~~~~~~~~~~
/home/andreas/apps/neumodvb/src/viewer/neumompv.cc:26:10: fatal error: 'GL/glut.h' file not found
#include <GL/glut.h>
```

**Comment by deeptho** [comment]
These should be the dependencies, based on the information gathered so far:
`
sudo  apt install -y  libboost-all-dev libgtk-3-0 libgtk-3-dev curl libcurl4-openssl-dev  libwxgtk-media3.0-gtk3-dev gettext libexif-dev libavcodec-dev libavformat-dev libavutil-dev libswscale-dev   python3-configobj python3-cachetools python3-jinja2 python3-pip clang-format python3-sip-dev libconfig-dev libconfig++-dev libdvbcsa-dev  libmpv-dev freeglut3-dev python-wxgtk3.0 python3-wxgtk-media4.0 python3-wxgtk-webview4.0 python3-wxgtk4.0 python3-scipy clang lsb-core lsb-release python3-regex liblog4cxx12 liblog4cxx-dev freeglut3
`
I will add this to the next version in the documentation

**Comment by andimik** [comment]
ok, thanks.

**Comment by andimik** [comment]
> These should be the dependencies, based on the information gathered so far:
sudo  apt install -y  libboost-all-dev libgtk-3-0 libgtk-3-dev curl libcurl4-openssl-dev  libwxgtk-media3.0-gtk3-dev gettext libexif-dev libavcodec-dev libavformat-dev libavutil-dev libswscale-dev   python3-configobj python3-cachetools python3-jinja2 python3-pip clang-format python3-sip-dev libconfig-dev libconfig++-dev libdvbcsa-dev  libmpv-dev freeglut3-dev python-wxgtk3.0 python3-wxgtk-media4.0 python3-wxgtk-webview4.0 python3-wxgtk4.0 python3-scipy clang lsb-core lsb-release python3-regex liblog4cxx12 liblog4cxx-dev freeglut3
I will add this to the next version in the documentation

No, this results in an error under Ubuntu 22.04 due to missing `python-wxgtk3.0` (which is no more available), but `libwxgtk3.0-gtk3-dev` is needed instead.

So, it should be 

```
sudo apt install -y libboost-all-dev libgtk-3-0 libgtk-3-dev curl libcurl4-openssl-dev  libwxgtk-media3.0-gtk3-dev gettext libexif-dev libavcodec-dev libavformat-dev libavutil-dev libswscale-dev   python3-configobj python3-cachetools python3-jinja2 python3-pip clang-format python3-sip-dev libconfig-dev libconfig++-dev libdvbcsa-dev  libmpv-dev freeglut3-dev libwxgtk3.0-gtk3-dev python3-wxgtk-media4.0 python3-wxgtk-webview4.0 python3-wxgtk4.0 python3-scipy clang lsb-core lsb-release python3-regex liblog4cxx12 liblog4cxx-dev freeglut3
```

**Comment by deeptho** [comment]
Fixed 5a572f37cad230

**Comment by andimik** [comment]
For Ubuntu 24.04.2 it seems that many packages have changed their name:

```
sudo apt install -y libboost-all-dev libgtk-3-dev curl libcurl4-openssl-dev  libwxgtk-media3.2-dev gettext libexif-dev libavcodec-dev libavformat-dev libavutil-dev libswscale-dev   python3-configobj python3-cachetools python3-jinja2 python3-pip clang-format python3-sip-dev libconfig-dev libconfig++-dev libdvbcsa-dev  libmpv-dev freeglut3-dev python3-wxgtk-media4.0 python3-wxgtk-webview4.0 python3-wxgtk4.0 python3-scipy clang  lsb-release python3-regex liblog4cxx15 liblog4cxx-dev
```

But the main problem is `cmake`, which you need to upgrade using the instruction on https://apt.kitware.com/
